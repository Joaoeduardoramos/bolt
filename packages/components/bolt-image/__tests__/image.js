const { render } = require('@bolt/twig-renderer');
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../image.schema.yml'));
const { tag } = schema.properties;

describe('<bolt-image> Component', async () => {
  test('<bolt-image> with ratio object compiles', async () => {
    const results = await render('@bolt-components-image/image.twig', {
      src: '/pattern-lab/images/placeholders/tout-4x3-climber.jpg',
      alt: 'A Rock Climber'
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('<bolt-image> with ratio object and lazyloading disabled compiles', async () => {
    const results = await render('@bolt-components-image/image.twig', {
      src: '/pattern-lab/images/placeholders/tout-4x3-climber.jpg',
      alt: 'A Rock Climber',
      lazyload: false,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
