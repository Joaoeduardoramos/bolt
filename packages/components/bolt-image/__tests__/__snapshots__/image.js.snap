// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-image> Component <bolt-image> with max-width and additional style attribute renders as expected 1`] = `
<bolt-image src="/fixtures/1200x660.jpg"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
            alt="A Rock Climber"
            max-width="200px"
            sizes="auto"
            ratio="1200/660"
            placeholder-color="rgb(190, 190, 190)"
            placeholder-image="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDIBCQkJDAsMGA0NGDIhHCEyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMv/CABEIAA0AGAMBIgACEQEDEQH/xAAXAAADAQAAAAAAAAAAAAAAAAAEBgcF/9oACAEBAAAAAHyeHVZAzv/EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQIQAAAAP//EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQMQAAAAP//EACUQAAEDAwIGAwAAAAAAAAAAAAIBAwQABQYREhMhIjEyYhRBov/aAAgBAQABPwDOY10et8aTa5TMcx6SN0d3IlHslcKTPvT5rdBdgIOnAUdFEqwozt2ZRwcdQvlNuMFy09kX81IjtS7K2D4IbZNJuFfvpq5Q46MOttNC0WvmPer1GC05NjrzHkQMGfsu/nX/xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AD//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AD//2Q=="
            style="display: block;background-color: rgb(190, 190, 190); width: 200px; max-width: 100%;"
>
  <bolt-ratio ratio="20/11">
    <replace-with-children class="c-bolt-ratio"
                           style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt="A Rock Climber"
           src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDIBCQkJDAsMGA0NGDIhHCEyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMv/CABEIAA0AGAMBIgACEQEDEQH/xAAXAAADAQAAAAAAAAAAAAAAAAAEBgcF/9oACAEBAAAAAHyeHVZAzv/EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQIQAAAAP//EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQMQAAAAP//EACUQAAEDAwIGAwAAAAAAAAAAAAIBAwQABQYREhMhIjEyYhRBov/aAAgBAQABPwDOY10et8aTa5TMcx6SN0d3IlHslcKTPvT5rdBdgIOnAUdFEqwozt2ZRwcdQvlNuMFy09kX81IjtS7K2D4IbZNJuFfvpq5Q46MOttNC0WvmPer1GC05NjrzHkQMGfsu/nX/xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AD//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AD//2Q=="
      >
      <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
           alt="A Rock Climber"
           data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
           data-sizes="auto"
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <noscript>
        <img class="c-bolt-image__image"
             alt="A Rock Climber"
             src="/fixtures/1200x660.jpg"
             srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
        >
      </noscript>
    </replace-with-children>
  </bolt-ratio>
</bolt-image>
`;

exports[`<bolt-image> Component <bolt-image> with max-width of 50% renders as expected 1`] = `
<bolt-image src="/fixtures/1200x660.jpg"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
            alt="A Rock Climber"
            max-width="50%"
            sizes="auto"
            ratio="1200/660"
            placeholder-color="rgb(190, 190, 190)"
            placeholder-image="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDIBCQkJDAsMGA0NGDIhHCEyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMv/CABEIAA0AGAMBIgACEQEDEQH/xAAXAAADAQAAAAAAAAAAAAAAAAAEBgcF/9oACAEBAAAAAHyeHVZAzv/EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQIQAAAAP//EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQMQAAAAP//EACUQAAEDAwIGAwAAAAAAAAAAAAIBAwQABQYREhMhIjEyYhRBov/aAAgBAQABPwDOY10et8aTa5TMcx6SN0d3IlHslcKTPvT5rdBdgIOnAUdFEqwozt2ZRwcdQvlNuMFy09kX81IjtS7K2D4IbZNJuFfvpq5Q46MOttNC0WvmPer1GC05NjrzHkQMGfsu/nX/xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AD//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AD//2Q=="
            style="background-color: rgb(190, 190, 190); width: 50%;"
>
  <bolt-ratio ratio="20/11">
    <replace-with-children class="c-bolt-ratio"
                           style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt="A Rock Climber"
           src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDIBCQkJDAsMGA0NGDIhHCEyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMv/CABEIAA0AGAMBIgACEQEDEQH/xAAXAAADAQAAAAAAAAAAAAAAAAAEBgcF/9oACAEBAAAAAHyeHVZAzv/EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQIQAAAAP//EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQMQAAAAP//EACUQAAEDAwIGAwAAAAAAAAAAAAIBAwQABQYREhMhIjEyYhRBov/aAAgBAQABPwDOY10et8aTa5TMcx6SN0d3IlHslcKTPvT5rdBdgIOnAUdFEqwozt2ZRwcdQvlNuMFy09kX81IjtS7K2D4IbZNJuFfvpq5Q46MOttNC0WvmPer1GC05NjrzHkQMGfsu/nX/xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AD//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AD//2Q=="
      >
      <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
           alt="A Rock Climber"
           data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
           data-sizes="auto"
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <noscript>
        <img class="c-bolt-image__image"
             alt="A Rock Climber"
             src="/fixtures/1200x660.jpg"
             srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
        >
      </noscript>
    </replace-with-children>
  </bolt-ratio>
</bolt-image>
`;

exports[`<bolt-image> Component <bolt-image> with max-width of 200px renders as expected 1`] = `
<bolt-image src="/fixtures/1200x660.jpg"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
            alt="A Rock Climber"
            max-width="200px"
            sizes="auto"
            ratio="1200/660"
            placeholder-color="rgb(190, 190, 190)"
            placeholder-image="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDIBCQkJDAsMGA0NGDIhHCEyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMv/CABEIAA0AGAMBIgACEQEDEQH/xAAXAAADAQAAAAAAAAAAAAAAAAAEBgcF/9oACAEBAAAAAHyeHVZAzv/EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQIQAAAAP//EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQMQAAAAP//EACUQAAEDAwIGAwAAAAAAAAAAAAIBAwQABQYREhMhIjEyYhRBov/aAAgBAQABPwDOY10et8aTa5TMcx6SN0d3IlHslcKTPvT5rdBdgIOnAUdFEqwozt2ZRwcdQvlNuMFy09kX81IjtS7K2D4IbZNJuFfvpq5Q46MOttNC0WvmPer1GC05NjrzHkQMGfsu/nX/xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AD//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AD//2Q=="
            style="background-color: rgb(190, 190, 190); width: 200px; max-width: 100%;"
>
  <bolt-ratio ratio="20/11">
    <replace-with-children class="c-bolt-ratio"
                           style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt="A Rock Climber"
           src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDIBCQkJDAsMGA0NGDIhHCEyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMv/CABEIAA0AGAMBIgACEQEDEQH/xAAXAAADAQAAAAAAAAAAAAAAAAAEBgcF/9oACAEBAAAAAHyeHVZAzv/EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQIQAAAAP//EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQMQAAAAP//EACUQAAEDAwIGAwAAAAAAAAAAAAIBAwQABQYREhMhIjEyYhRBov/aAAgBAQABPwDOY10et8aTa5TMcx6SN0d3IlHslcKTPvT5rdBdgIOnAUdFEqwozt2ZRwcdQvlNuMFy09kX81IjtS7K2D4IbZNJuFfvpq5Q46MOttNC0WvmPer1GC05NjrzHkQMGfsu/nX/xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AD//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AD//2Q=="
      >
      <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
           alt="A Rock Climber"
           data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
           data-sizes="auto"
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <noscript>
        <img class="c-bolt-image__image"
             alt="A Rock Climber"
             src="/fixtures/1200x660.jpg"
             srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
        >
      </noscript>
    </replace-with-children>
  </bolt-ratio>
</bolt-image>
`;

exports[`<bolt-image> Component <bolt-image> with ratio object and lazyloading disabled compiles 1`] = `
<bolt-image src="/fixtures/1200x660.jpg"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
            alt="A Rock Climber"
            sizes="auto"
            ratio="1200/660"
            no-lazy
>
  <bolt-ratio ratio="20/11">
    <replace-with-children class="c-bolt-ratio"
                           style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image"
           alt="A Rock Climber"
           srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
           sizes="auto"
           src="/fixtures/1200x660.jpg"
      >
    </replace-with-children>
  </bolt-ratio>
</bolt-image>
`;

exports[`<bolt-image> Component <bolt-image> with ratio object compiles 1`] = `
<bolt-image src="/fixtures/1200x660.jpg"
            srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
            alt="A Rock Climber"
            sizes="auto"
            ratio="1200/660"
            placeholder-color="rgb(190, 190, 190)"
            placeholder-image="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDIBCQkJDAsMGA0NGDIhHCEyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMv/CABEIAA0AGAMBIgACEQEDEQH/xAAXAAADAQAAAAAAAAAAAAAAAAAEBgcF/9oACAEBAAAAAHyeHVZAzv/EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQIQAAAAP//EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQMQAAAAP//EACUQAAEDAwIGAwAAAAAAAAAAAAIBAwQABQYREhMhIjEyYhRBov/aAAgBAQABPwDOY10et8aTa5TMcx6SN0d3IlHslcKTPvT5rdBdgIOnAUdFEqwozt2ZRwcdQvlNuMFy09kX81IjtS7K2D4IbZNJuFfvpq5Q46MOttNC0WvmPer1GC05NjrzHkQMGfsu/nX/xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AD//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AD//2Q=="
            style="background-color: rgb(190, 190, 190);"
>
  <bolt-ratio ratio="20/11">
    <replace-with-children class="c-bolt-ratio"
                           style="padding-bottom: calc((11 / 20) * 100%); --aspect-ratio: 1.8181818181818;"
    >
      <img class="c-bolt-image__image-placeholder"
           alt="A Rock Climber"
           src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDIBCQkJDAsMGA0NGDIhHCEyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMv/CABEIAA0AGAMBIgACEQEDEQH/xAAXAAADAQAAAAAAAAAAAAAAAAAEBgcF/9oACAEBAAAAAHyeHVZAzv/EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQIQAAAAP//EABQBAQAAAAAAAAAAAAAAAAAAAAD/2gAIAQMQAAAAP//EACUQAAEDAwIGAwAAAAAAAAAAAAIBAwQABQYREhMhIjEyYhRBov/aAAgBAQABPwDOY10et8aTa5TMcx6SN0d3IlHslcKTPvT5rdBdgIOnAUdFEqwozt2ZRwcdQvlNuMFy09kX81IjtS7K2D4IbZNJuFfvpq5Q46MOttNC0WvmPer1GC05NjrzHkQMGfsu/nX/xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AD//xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAEDAQE/AD//2Q=="
      >
      <img class="c-bolt-image__image c-bolt-image__lazyload c-bolt-image__lazyload--fade c-bolt-image__lazyload--blur js-lazyload"
           alt="A Rock Climber"
           data-srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
           data-sizes="auto"
           src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
      >
      <noscript>
        <img class="c-bolt-image__image"
             alt="A Rock Climber"
             src="/fixtures/1200x660.jpg"
             srcset="/fixtures/1200x660-50.jpg 50w, /fixtures/1200x660-100.jpg 100w, /fixtures/1200x660-200.jpg 200w, /fixtures/1200x660-320.jpg 320w, /fixtures/1200x660-480.jpg 480w, /fixtures/1200x660-640.jpg 640w, /fixtures/1200x660-800.jpg 800w, /fixtures/1200x660-1024.jpg 1024w"
        >
      </noscript>
    </replace-with-children>
  </bolt-ratio>
</bolt-image>
`;
